free-1.md

```diff
@@ -9,42 +9,141 @@ good way to meet their needs.
+usable in popular word processors? Regardless, I would imagine some note snippets
+to look similar to the following in this language:
+
+``` 
+This is a standard supply and demand graph
+
+Graph{
+	Y-Axis: Price
+	X-Axis: Quantity
+	SupplyCurve()	
+	DemandCurve()
+}
+
+generate Graph
+```
+
+```
+Today, we're learning about the *Engel Curve*, the graph looks like this: 
+
+Graph{
+	Y-axis: Income/month
+	X-axis: Units
+	EngelCurve()
+}
+
+generate Graph
+```	
+
+By instantiating some Graph object, you can generate a graph in place in the 
+notes. Then the user could label axes and specify the curves that are needed 
+in the graph. So you could specify SupplyCurve(), which would give a default
+upward sloping supply curve. Each curve could take certain parameters for
+transformations, for example `SupplyCurve(Slope: .5, Y: -10, X: 50  )`, would
+give you a supply curve with slope of .5 starting at a Y-value of -10 and X-value
+of 50. Also, we'd want to support markdown in the language to give users
+easy formatting options when writing notes.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
+The program would ideally run on save.  When it runs, the language would generate
+a graph image and then copy it to the clipboard for the user to paste or
+ideally make it appear in place, but this might be difficult. An early issue I 
+could forsee happening is if a user specifies a curve that is not supported in
+the program.  For example, if they specified `ProductionFunction()` and we didn't
+support that curve, there would be an error.  The way for this error to appear
+would likely be through some sort of linter, much like other programming
+languages have. We would then notify users at compile time, or in this case on
+save that there is an issue and communicate it clearly.
 
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
-
+It should be easy to both take notes in natural language and generate graphs in
+the same editor or workspace. It should be possible to create extremely custom
+graphs, but I imagine that it would be difficult and may require more time
+than available during a class session.  It should be impossible to do anything
+other than take notes in natural language or create economics graphs. No web
+development, solving algorithms homework, etc.
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well 
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
-
+There aren't any DSLs that do this in the editor. However, there are tools
```

Comment:
It would be cool if you could take hand-drawn graphs, as in OneNote, and somehow convert them to clean, generated graphs.